src/dxvk/dxvk_extensions.h

```diff
@@ -343,6 +343,14 @@ namespace dxvk {
     DxvkExt extSurfaceMaintenance1          = { VK_EXT_SURFACE_MAINTENANCE_1_EXTENSION_NAME,            DxvkExtMode::Optional };
     DxvkExt khrGetSurfaceCapabilities2      = { VK_KHR_GET_SURFACE_CAPABILITIES_2_EXTENSION_NAME,       DxvkExtMode::Optional };
     DxvkExt khrSurface                      = { VK_KHR_SURFACE_EXTENSION_NAME,                          DxvkExtMode::Required };
+
+    /**
+     * \brief Get list of supported extensions
+     * \returns List of supported extensions
+     */
+    std::initializer_list<const DxvkExt*> getExtensionList() const {
```

Comment:
This is UB, need to return a proper container here.

Would also prefer this method to be moved to `DxvkInstance` so that it actually and unambiguously returns the enabled extensions.

Reply:
Oops, that was supposed to be a vector. Fixed it. It doesn't return the enabled extensions, just a list of all `DxvtExt` objects in `DxvkInstanceExtensions` that can be queried for their enabled status. The point of putting it in `DxvkInstanceExtensions` was that if someone adds a new extension then they'll know to add it to the list since it's right there in the struct.

Reply:
We already have similar code for device extensions in `DxvkAdapter`. I don't want this done in two different ways for no reason.

Reply:
Okay, I'll rewrite it.

Reply:
With b436477, we now return the extension name list that the instance was created with. This fixes extensions from extension providers not being included and entirely avoids the problem of having to change the code if new extensions are supported.